server/controllers/convo.js

```diff
@@ -0,0 +1,81 @@
+const express = require("express");
+const router = express.Router();
+const mongoose = require("mongoose");
+mongoose.Promise = global.Promise;
+const Convo = require("../models/convo");
+
+// POST operation -- Create Convo
+module.exports.createConvo = async (req, res, next) => {
+    const { users } = req.body;
+    const alreadyExists = await Convo.findOne({
+        users: users,
+    });
+    if (users.length >= 1) {
+        if (!alreadyExists) {
+            const newConvo = new Convo({
+                users: users,
+            });
+            try {
+                newConvo.save().then((err) => {
+                    if (err) {
+                        console.log(err);
+                        res.send(err);
+                    } else {
+                        res.send(201);
+                        res.json(newConvo._id);
+                    }
+                });
+            } catch (err) {
+                res.status(400).send(err);
+            }
+        } else {
+            res.status(400);
+            res.send("Conversation between these users already exists.");
+        }
+    } else {
+        res.status(400);
+        res.send("Conversation must include more than one user in user array.");
+    }
+};
+
+// GET -- Fetch all convos
+module.exports.getAllConvos = async (req, res, next) => {
+    try {
+        Convo.find({}, (err, convos) => {
+            res.send(convos).status(200);
+        });
```

Comment:
This would ultimately expose conversations to people that shouldn't have access to it. I would suggest filtering the request based on the conversations the authenticated user has access to.

Reply:
Modified to only return conversations that the user is a member of.

Reply:
I realized that fetching a single convo should also check for the user being a member to that convo -- I modified getSingleConvo to include this, too.